samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -134,6 +136,24 @@ public LocalApplicationRunner(Config config) {
   }
 
   @Override
+  public void runTask() {
```

Comment:
Can we not make the interface runTask(TaskFactory) ? That way, we can avoid class-loading from config and users can easily provide a taskfactory that produces a StreamTask or AsynsStreamTask. This will make it easier to integrate with dependency injection systems (like offspring) and it will look more like a library. 
I understand that StreamTask or AsyncStreamTask are two independent interfaces. We should be able to inspect that easily.